packages/demux/src/AloxideActionHandler.ts

```diff
@@ -143,6 +168,52 @@ export class AloxideActionHandler extends AbstractActionHandler {
     return handle(null, context);
   }
 
+  public async handleBlock(nextBlock: NextBlock, isReplay: boolean): Promise<number | null> {
+    try {
+      return await super.handleBlock(nextBlock, isReplay);
+    } catch (e) {
+      if (e instanceof MismatchedBlockHashError) {
+        if (this.lastProcessedBlockHash) {
+          /* ActionReader store processed block info in ephemeral memory,
+            in case of they lost it, they don't can detect a fork and return new block to ActionHandler
+            ActionHandler store lasted index state in DB, detect that fork but can detect the point of fork is
+            ActionHandler should rollback to last irreversible blockNumber
+            Ref https://github.com/EOSIO/demux-js/issues/123#issuecomment-465756166
+          */
+
+          // - rollbackTo should be as fast as possible to speed of sync with blockchain
+          await this.handleRollbackToLastIreversibleBlock();
+          return this.lastIrreversibleBlockNumber + 1;
+        } else {
+          const handleWithArgs: (state: any, context?: any) => Promise<void> = async (
+            state: any,
+            context: any = {},
+          ) => {
+            await this.handleActions(state, context, nextBlock, isReplay);
+          };
+          await this.handleWithState(handleWithArgs);
+          return null;
+        }
+      }
+
+      throw e;
+    }
+  }
+
+  private async handleRollbackToLastIreversibleBlock() {
```

Comment:
why does this need to be a function alone?

Reply:
this function create debug log about rollback, call function to roll back data and roll back deferred effects.